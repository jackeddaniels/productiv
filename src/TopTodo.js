import React from "react";

import Todo from "./Todo";

/** Shows the top todo.
 *
 * Props:
 * - todos
 *
 * TodoApp -> TopTodo
 */

function TopTodo({todos}) {
  // lowest-priority # is the highest priority
  let top = todos.reduce(
      (acc, cur) => cur.priority < acc.priority ? cur : acc, todos[0]);

  return (
  <div>
    {top && <Todo todo={top}/>}
    {!top && <p>No top todo available.</p>}

  </div>
  );
}

export default TopTodo;