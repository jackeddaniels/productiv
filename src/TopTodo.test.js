import React from "react";
import { render } from "@testing-library/react";
import TopTodo from "./TopTodo";

describe("TopTodo component", function () {
  it("renders without crashing", function () {
    render(<TopTodo todos={[{
      id: 1,
      title: "Code!",
      description: "Write some code",
      priority: 2,
    }]}/>);
  });

  it("matches snapshot", function () {
    const { container } = render(<TopTodo todos={[{
      id: 1,
      title: "Code!",
      description: "Write some code",
      priority: 2,
    }]}/>);
    expect(container).toMatchSnapshot();
  });
});