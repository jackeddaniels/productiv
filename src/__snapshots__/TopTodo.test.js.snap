// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopTodo component matches snapshot 1`] = `
<div>
  <div>
    <div
      class="Todo"
    >
      <div>
        <b>
          Code!
        </b>
         
        <small>
          (priority: 
          2
          )
        </small>
      </div>
      <div>
        <small>
          Write some code
        </small>
      </div>
    </div>
  </div>
</div>
`;
